tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_interactive_view.ts

```diff
@@ -0,0 +1,450 @@
+  templateUrl: './line_chart_interactive_view.ng.html',
+  styleUrls: ['./line_chart_interactive_view.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: RepositionScrollStrategy,
+      useFactory: scrollStrategyFactory,
+      deps: [Overlay],
+    },
+  ],
+})
+export class LineChartInteractiveViewComponent implements OnChanges, OnDestroy {
+  @ViewChild('dots', {static: true, read: ElementRef})
+  dotsContainer!: ElementRef<SVGElement>;
+
+  @ViewChild(CdkConnectedOverlay)
+  overlay!: CdkConnectedOverlay;
+
+  @Input()
+  seriesData!: DataSeries[];
+
+  @Input()
+  seriesMetadataMap!: DataSeriesMetadataMap;
+
+  @Input()
+  viewExtent!: Extent;
+
+  @Input()
+  xScale!: Scale;
+
+  @Input()
+  yScale!: Scale;
+
+  @Input()
+  domDim!: Dimension;
+
+  @Input()
+  tooltipOriginEl!: ElementRef;
+
+  @Input()
+  tooltipTemplate?: TooltipTemplate;
+
+  @Output()
+  onViewExtentChange = new EventEmitter<{dataExtent: Extent}>();
+
+  @Output()
+  onViewExtentReset = new EventEmitter<void>();
+
+  readonly InteractionState = InteractionState;
+
+  state: InteractionState = InteractionState.NONE;
+
+  // Gray box that shows when user drags with mouse down
+  zoomBoxInUiCoordinate: Rect = {x: 0, width: 0, height: 0, y: 0};
+
+  readonly tooltipPositions: ConnectedPosition[] = [
+    // Prefer align at bottom edge of the line chart
+    {
+      offsetY: 5,
+      originX: 'start',
+      overlayX: 'start',
+      originY: 'bottom',
+      overlayY: 'top',
+    },
+    // Then top
+    {
+      offsetY: 5,
+      originX: 'start',
+      overlayX: 'start',
+      originY: 'top',
+      overlayY: 'bottom',
+    },
+    // then right
+    {
+      offsetX: 5,
+      originX: 'end',
+      overlayX: 'start',
+      originY: 'top',
+      overlayY: 'top',
+    },
+  ];
+
+  cursorXLocation: number | null = null;
+  cursoredData: TooltipDatum[] = [];
+  tooltipDisplayAttached: boolean = false;
+
+  @HostBinding('class.show-zoom-instruction')
+  showZoomInstruction: boolean = false;
+
+  private dragStartCoord: {x: number; y: number} | null = null;
+  private isCursorInside = false;
+  private readonly ngUnsubscribe = new Subject();
+
+  constructor(
+    private readonly changeDetector: ChangeDetectorRef,
+    readonly scrollStrategy: RepositionScrollStrategy
+  ) {}
+
+  ngAfterViewInit() {
+    fromEvent<MouseEvent>(this.dotsContainer.nativeElement, 'dblclick', {
+      passive: false,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        // Prevent double click from selecting text.
+        event.preventDefault();
+        this.onViewExtentReset.emit();
+        this.state = InteractionState.NONE;
+        this.changeDetector.markForCheck();
+      });
+
+    fromEvent<MouseEvent>(this.dotsContainer.nativeElement, 'mousedown', {
+      passive: false,
+    })
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe((event) => {
+        event.preventDefault();
```

Comment:
Similarly here, why do we need to preventDefault on mousedown?  If there is a good reason, can we add a comment?